test/script_tools.py

```diff
@@ -146,3 +147,15 @@ def test_opts(self, prol):
         logging.debug("options wo default sync options %s", ms.options)
         self.assertEqual(ms.options.__dict__, myopts)
 
+    @mock.patch('vsc.utils.script_tools.ExtendedSimpleOption.prologue')
+    @mock.patch('vsc.utils.nagios.ok_exit')
+    @mock.patch('vsc.utils.nagios.warning_exit')
+    @mock.patch('vsc.utils.nagios.unknown_exit')
+    @mock.patch('vsc.utils.nagios.critical_exit')
+    def test_exit(self, mock_critical, mock_unknown, mock_warning, mock_ok, mock_prologue):
+
+        cli = MyCLI()
+
+        cli.warning("be warned")
+        #mock_warning.is_called()
```

Comment:
why don;t you mock `exit_from_errorcode` and see if it is called correctly?

Reply:
Because that would actually mean that I need to reproduce what it looks like here, which kinda circumvents the test. I want/need to just be able to call it.

Reply:
If I would check how it is called, I can just call it in the right way :-p

Reply:
the only thing you should test here if is you use the api correctly (looking at the commit, it was wrong before). if want to test the api, add tests for `vsc.utils.nagios`.

Reply:
No, I called `exit_from_errorcode` in the wrong way. How would testing the nagios.py API help with that?

Reply:
not at all. i'm only suggesting that you should only test that it is called correctly. checking if warning is called, only tests if exit_from_errorcode calls warning correctly if you call exit_from_errorcode correctly.

Reply:
If it is incorrectly called, it will result in an unknown exit, hence have the wrong second argument.